### ISSUE TITLE
Solid Queue no longer starts with rails/server in the production env

### ISSUE THREAD

**Issue by kirillshevch** [rails/solid_queue#281]
Hi folks!

After the version bump from 0.3 to 0.5, Solid Queue no longer starts while starting Rails process in the production environment. Is this expected behavior and should it now be treated exclusively as a separate run via `bundle exec rake solid_queue:start`? It was quite handy to run it all together for small projects.

I didn't find anything regarding that in UPGRADING.md

Thanks!

**Comment by rosa** [comment]
Hey @kirillshevch, that's odd! Were you using the Puma plugin before? That was the only way for Solid Queue to start automatically. That hasn't changed. What configuration do you have?

**Comment by kirillshevch** [comment]
Hi @rosa,
Yes, I was using `plugin :solid_queue` in puma config, no changes there. Rails starts in Docker container `CMD ["./bin/rails", "server"]`

**Comment by rosa** [comment]
Thanks @kirillshevch! What's your Solid Queue configuration?

**Comment by gugatxr** [comment]
Hi, I have the same problem. 


puma version: 6.4.2
rails version: 7.1.3.4
solid_queue: 0.6.0
ruby: 3.3.0

Using `plugin :solid_queue`

My config: 

```
default: &default
  dispatchers:
    - polling_interval: 1
      batch_size: 500
  workers:
    - queues: "*"
      threads: 3
      processes: 1
      polling_interval: 0.1

development:
 <<: *default

test:
 <<: *default

production:
 <<: *default
```

**Comment by rosa** [comment]
Is this happening only in production? Does it still work in development?

**Comment by gugatxr** [comment]
I was able to fix my problem.
In my case, I have a Hetzner server with 2 vCPUs. By default, the Puma workers setting was set to 1. Setting it to 2 workers resolved the issue.

**Comment by rosa** [comment]
Thanks a lot for letting us know, @gugatxr 🙏 

Unfortunately, we're going to stop including the Puma plugin with Solid Queue: https://github.com/rails/solid_queue/pull/303. You can still use it but it won't be included with the gem. 

@kirillshevch, I suspect your problem might be related to your Solid Queue configuration, since the upgrades did some changes there so that in some cases a default config won't be used where it was being provided before, and that might be causing the Solid Queue process to crash. If you could share your configuration, that would help confirm if that's what's happening here.

**Comment by kirillshevch** [comment]
Is there a reference on how to use the Puma plugin since it is no longer included?

**Comment by rosa** [comment]
@kirillshevch yes! It's still included but it'll go away in version 1.0. When that happens, I've kept it around in the dummy app for testing and also linked to a gist with it in the README. Check https://github.com/rails/solid_queue/pull/303 for all the details.

**Comment by dhh** [comment]
We ended up reverting this. The puma plugin will stay for single-server deployments. You can see how Rails 8 configures it behind an ENV flag in config/puma.rb here: https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/config/puma.rb.tt#L38

So you can do that yourself in puma.rb as well.
